src/otx/algo/utils/xai_utils.py

```diff
@@ -67,40 +92,43 @@ def process_saliency_maps(
     return processed_saliency_maps
 
 
-def convert_maps_to_dict_all(saliency_maps: np.array) -> list[dict[Any, np.array]]:
+def convert_maps_to_dict_all(saliency_map: list[np.ndarray]) -> list[dict[Any, np.array]]:
     """Convert salincy maps to dict for TargetExplainGroup.ALL."""
-    if saliency_maps[0].ndim != 3:
-        msg = "Shape mismatch."
-        raise ValueError(msg)
-
     processed_saliency_maps = []
-    for maps_per_image in saliency_maps:
+    for maps_per_image in saliency_map:
+        if maps_per_image.ndim != 3:
+            msg = "Shape mismatch."
+            raise ValueError(msg)
+
```

Comment:
There's no need to move this check in `for` loop since all saliency maps should be the same size based on the method of its generation. But your option looks prettier.

Reply:
The former one produces an error not understandable easily if `saliency_maps` is an empty list.

Reply:
Makes sense